src/rdpBuilder.js

```diff
@@ -0,0 +1,177 @@
+                          'negotiate security layer:i:1\n' +
+                          'remoteapplicationmode:i:0\n' +
+                          'alternate shell:s:\n' +
+                          'shell working directory:s:\n' +
+                          'gatewayhostname:s:\n' +
+                          'gatewayusagemethod:i:4\n' +
+                          'gatewaycredentialssource:i:4\n' +
+                          'gatewayprofileusagemethod:i:0\n' +
+                          'promptcredentialonce:i:1\n' +
+                          'use redirection server name:i:0'
+export default class RDPBuilder {
+  constructor (vmName, username) {
+    this.address = vmName
+    this.guestID = ''
+    this.fullScreen = true
+    this.fullScreenTitle
+    this.width = 640
+    this.height = 480
+    this.authenticationLevel = 2
+    this.useLocalDrives
+    this.redirectPrinters = false
+    this.redirectClipboard = true
+    this.redirectSmartCards = false
+    this.username = username
+  }
+
+  getUsername () {
+    return this.username
+  }
+
+  getFullScreen () {
+    return this.fullScreen
+  }
+
+  setFullScreen (fullScreen) {
+    this.fullScreen = fullScreen
+  }
+
+  getWidth () {
+    return this.width
+  }
+
+  setWidth (width) {
+    this.width = width
+  }
+
+  getHeight () {
+    return this.height
+  }
+
+  setHeight (height) {
+    this.height = height
+  }
+
+  getAuthenticationLevel () {
+    return this.authenticationLevel
+  }
+
+  setAuthenticationLevel (authenticationLevel) {
+    this.authenticationLevel = authenticationLevel
+  }
+
+  getRedirectPrinters () {
+    return this.redirectPrinters
+  }
+
+  setRedirectPrinters (redirectPrinters) {
+    this.redirectPrinters = redirectPrinters
+  }
+
+  getRedirectClipboard () {
+    return this.redirectClipboard
+  }
+
+  setRedirectClipboard (redirectClipboard) {
+    this.redirectClipboard = redirectClipboard
+  }
+
+  getRedirectSmartCards () {
+    return this.redirectSmartCards
+  }
+
+  setRedirectSmartCards (redirectSmartCards) {
+    this.redirectSmartCards = redirectSmartCards
+  }
+
+  getAddress () {
+    return this.address
+  }
+
+  setAddress (value) {
+    this.address = value
+  }
+
+  getGuestID () {
+    return this.guestID
+  }
+
+  setGuestID (value) {
+    this.guestID = value
+  }
+
+  getUseLocalDrives () {
+    return this.useLocalDrives
+  }
+
+  setUseLocalDrives (value) {
+    this.useLocalDrives = value
+  }
+
+  booleanToInt (b) {
+    if (b == null || b === false) {
+      return 0
+    }
+
+    return 1
+  }
+
+  getScreenMode () {
+    if (this.getFullScreen()) {
+      return 2
+    }
+
+    return 1
+  }
+
+  getredirectDrivesLines () {
+    if (this.getUseLocalDrives()) {
+      return '\ndrivestoredirect:s:*'
+    } else {
+      return '\ndrivestoredirect:s:'
+    }
+  }
+
+  getEnableCredSspSupport () {
+    return true
+  }
+
+  buildRDP () {
+    let result = BASE_CONFIG_FILE
+    result += `\nscreen mode id:i:${this.getScreenMode()}` +
```

Comment:
A ``` `template string` ``` can span over multiple lines, so you can elide all those sring additions. Also consider removing the very first string addition by using 
```javascript
return `${BASE_CONFIG_FILE}
screen mode id:....
...
...`
````
Notice the newlines will now be taken literally and no need to state them again